src/components/Pagination/Pagination.js

```diff
@@ -52,57 +51,57 @@ export class RawPagination extends Component {
     );
   }
 
-  previousPageLink(pager, createURL) {
+  previousPageLink({ isFirstPage, currentPage, createURL }) {
     return this.pageLink({
       ariaLabel: 'Previous',
       additionalClassName: this.props.cssClasses.previous,
-      isDisabled: this.props.nbHits === 0 || pager.isFirstPage(),
+      isDisabled: this.props.nbHits === 0 || isFirstPage,
       label: this.props.labels.previous,
-      pageNumber: pager.currentPage - 1,
+      pageNumber: currentPage - 1,
       createURL,
     });
   }
 
-  nextPageLink(pager, createURL) {
+  nextPageLink({ isLastPage, currentPage, createURL }) {
     return this.pageLink({
       ariaLabel: 'Next',
       additionalClassName: this.props.cssClasses.next,
-      isDisabled: this.props.nbHits === 0 || pager.isLastPage(),
+      isDisabled: this.props.nbHits === 0 || isLastPage,
       label: this.props.labels.next,
-      pageNumber: pager.currentPage + 1,
+      pageNumber: currentPage + 1,
       createURL,
     });
   }
 
-  firstPageLink(pager, createURL) {
+  firstPageLink({ isFirstPage, createURL }) {
     return this.pageLink({
       ariaLabel: 'First',
       additionalClassName: this.props.cssClasses.first,
-      isDisabled: this.props.nbHits === 0 || pager.isFirstPage(),
+      isDisabled: this.props.nbHits === 0 || isFirstPage,
       label: this.props.labels.first,
       pageNumber: 0,
       createURL,
     });
   }
 
-  lastPageLink(pager, createURL) {
+  lastPageLink({ isLastPage, total, createURL }) {
     return this.pageLink({
       ariaLabel: 'Last',
       additionalClassName: this.props.cssClasses.last,
-      isDisabled: this.props.nbHits === 0 || pager.isLastPage(),
+      isDisabled: this.props.nbHits === 0 || isLastPage,
       label: this.props.labels.last,
-      pageNumber: pager.total - 1,
+      pageNumber: total - 1,
       createURL,
     });
   }
 
-  pages(pager, createURL) {
-    const pages = [];
+  pages({ currentPage, pages, createURL }) {
+    const _pages = [];
 
-    forEach(pager.pages(), pageNumber => {
-      const isActive = pageNumber === pager.currentPage;
+    forEach(pages, pageNumber => {
```

Comment:
Not sure about that, but as soon as we refactor some code we can begin to get rid of the `lodash` functions like `forEach`, `map`, `reduce`, ... 

cc @bobylito

Reply:
We can as long as we stay opportunistic.

Reply:
If pages is an array, we should go for using the native method as suggested by @samouss